### ISSUE TITLE
Ability to download logs

### ISSUE THREAD

**Issue by PMunch** [AdguardTeam/AdGuardHome#3285]
### Problem Description
I find myself wondering about the DNS traffic on my network. And I would really like to download the full log of DNS queries/replies in a format that I can process. The log on the website is okay, but I'm not really able to search for specific record types etc. so having the full log in a script-friendly format would be nice.

### Proposed Solution
Output the log as JSON, CSV, or a custom format

### Alternatives Considered
Scraping the log on the web interface, but that seems a bit counter to the idea of running my own DNS server under HomeAssistant (which is how I'm using AdGuard Home). Unrelated it would also be nice if some AdGuard Home information could be made accessible to other HomeAssistant components. For example the amount of blocked queries, the total amount of queries etc. so that these could be used as part of an interface or to trigger rules.

**Comment by ainar-g** [comment]
Hi, thanks for the feature request.  This will probably depend on #2290, so it won't land earlier than that one, but once it's done, something like that should be relatively easy.

**Comment by PMunch** [comment]
How is the log stored now? Could there just be a download raw button with whatever format it is in now?

**Comment by Aikatsui** [comment]
> How is the log stored now? Could there just be a download raw button with whatever format it is in now?

See `querylog.json' file in the adguardhome folder

**Comment by Aikatsui** [comment]
Duplicate of #1176

**Comment by ainar-g** [comment]
I think @Aikatsui is correct, we will merge this one into the original issue.  Thanks.
